internal/hcsoci/resources.go

```diff
@@ -162,12 +145,28 @@ func ReleaseResources(ctx context.Context, r *Resources, vm *uvm.UtilityVM, all
 		r.scsiMounts = nil
 	}
 
+	if len(r.layers) != 0 {
+		op := UnmountOperationSCSI
+		if vm == nil || all {
+			op = UnmountOperationAll
+		}
+		var crp string
+		if vm != nil {
+			crp = containerRootfsPath(vm, r.containerRootInUVM)
+		}
+		err := UnmountContainerLayers(ctx, r.layers, crp, vm, op)
+		if err != nil {
+			return err
+		}
+		r.layers = nil
+	}
+
```

Comment:
Note: this reordering is necessary to properly unmount the scratch path when the container has mounts. Otherwise, unmounting the scratch path gives a "device or resource busy" error.

Reply:
Oh right because mounts are now within the scratch